### PR TITLE
test: improve coverage

### PR DESCRIPTION
**Pull request by SukkaW**
## What does it do?

This PR is a part of #3984

## How to test

```sh
git clone -b test-coverage-improve-1 https://github.com/sukkaw/hexo.git
cd hexo
npm install
npm test
```

## Pull request tasks

- [ ] Add test cases for the changes.
- [x] Passed the CI test.

### CODE REVIEW

File: lib/extend/tag.js [hunk 1]
```diff
@@ -214,12 +214,7 @@ class Tag {
     this.env.addExtension(name, tag);
   }
 
-  render(str, options, callback) {
-    if (!callback && typeof options === 'function') {
-      callback = options;
-      options = {};
-    }
-
+  render(str, options = {}) {
```
Comment (SukkaW) [inline lib/extend/tag.js]: `Tag.render()` just doesn't have callback, so I remove those lines and `callback` parameter directly.

File: test/scripts/console/deploy.js [hunk 2]
```diff
@@ -18,6 +18,12 @@ describe('deploy', () => {
 
   after(() => fs.rmdir(hexo.base_dir));
 
+  it('no deploy config', () => {
+    delete hexo.config.deploy;
+
+    return deploy({test: true});
```
Comment (curbengh) [inline test/scripts/console/deploy.js]: What is the expected return value of `deploy()`, after it resolved?
suggest to refactor to async/await (including other tests in this PR).

File: lib/extend/tag.js (hunk 1 above)
Comment (curbengh) [reply lib/extend/tag.js]: Seems like a callback is promisified?

https://github.com/hexojs/hexo/blob/640b91e4b13d37a9fbdaf518a296354345bdf7ff/lib/extend/tag.js#L229

File: test/scripts/console/deploy.js (hunk 2 above)
Comment (SukkaW) [reply test/scripts/console/deploy.js]: > What is the expected return value of `deploy()`, after it resolved?

No value at all. When there is no config, `console.log()` will be used to display an error message.

**Pull request merged**